wiring/src/spark_wiring_ble.cpp

```diff
@@ -1192,13 +1593,60 @@ int BleLocalDevice::on() {
 
 int BleLocalDevice::off() {
     WiringBleLock lk;
+    CHECK(hal_ble_stack_deinit(nullptr));
```

Comment:
Good question! Yes, it will prevent the BLE control request from working. How about simply returning `SYSTEM_ERROR_NOT_SUPPORTED` for now, until we come up a method to deal with the conflict?

Reply:
So, I've noticed we don't have `begin()` and `end()` for the `BleLocalDevice`. Why don't we add those and for now simply call `end()` in here to disable anything configured by the application or it's not possible to at the moment?

Reply:
It depends on what is done in the `end()` function. To make sure the BLE control request work as expected, neither disconnecting nor stopping advertising should happen. In that case, that is also weird  as being an `end()` function.

Reply:
> neither disconnecting nor stopping advertising should happen

I'd say both should happen though: stopping application-specific advertisement and terminating any application-initiated/handled connections.

Reply:
Sounds like a plan!

Reply:
If we added the `begin()` and `end()` functions, the `begin()` functions actually does nothing. So it seams a bit over designed. How about introducing a bool argument for the `off()` API to alternatively completely turn off the BLE radio or just terminate the BLE activities initiated by user application?

Reply:
I definitely wouldn't add any arguments to `off()` as to keep the API consistent with other things we already have. If we want to 'terminate the BLE activities initiated by user application' it needs to be `end()`. It's also fine if either `on()` or `begin()` is almost a no-op.

Reply:
Cool. Will implement that in a separated PR.